api/services/user_program_service.py

```diff
@@ -0,0 +1,20 @@
+from api import models, helpers, db
+
+
+class UserProgramService(object):
+    def __init__(self):
+        self.user_id = None
+        self.user_phone = None
+        self.user_program_data = None
+
+    def set_init_data(self, jsonData):
+        user_phone = helpers.fetch_by_key("urn", jsonData["contact"])
+        self.user_phone = helpers.sanitize_phone_string(user_phone)
+        user = models.User.query.get_by_phone(self.user_phone)
+        self.user_id = user.id
+
+    def mark_user_program_as_completed(self, JsonData):
+        self.set_init_data(JsonData)
+        self.user_program_data = models.UserProgram.get_by_user_id(self.user_id)
```

Comment:
This will be futuristic, but what will happen in the case of multiple user experiences?

Reply:
get_by_user_id function give use the latest active (i.e status = in-progress) user_program.